compiler-core/src/config.rs

```diff
@@ -818,6 +820,26 @@ mod uri_serde {
     }
 }
 
+// This is used for optional URIs
+mod option_uri_serde {
+    use serde::{Deserialize, Deserializer};
+    use super::uri_serde;
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Option<http::Uri>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let string: Option<String> = Option::deserialize(deserializer)?;
+        match string {
+            Some(s) => {
+                let deserializer = serde::de::value::StringDeserializer::new(s);
+                uri_serde::deserialize(deserializer).map(Some)
+            },
+            None => Ok(None),
+        }
+    }
+}
```

Comment:
How come a custom decoder is needed here?

Reply:
Mostly due to inexperience. Rust isn't my strong suite, and admittedly this was the path of least resistance for getting tests to pass. I can spend a bit more time on it this evening and attempt to merge decoding `Option<Uri>` into the `uir_serde` decoder if that's preferable.

Reply:
Have just removed the `option_uri_serde` decoder, instead merging the `Option<Uri>` logic into the base `uri_serde` decoder. Thanks for the nudge to clean that up. I wasn't totally happy making a new decoder for this case 😅